_episodes/05-loop.md

```diff
@@ -428,98 +428,142 @@ so she decides to get some coffee and catch up on her reading.
 
 > ## Variables in Loops
 >
-> Suppose that `ls` initially displays:
+> This exercise refers to the data-shell/molecules directory.
+> `ls` gives the following output:
 >
 > ~~~
-> fructose.dat    glucose.dat   sucrose.dat
+> cubane.pdb  ethane.pdb  methane.pdb  octane.pdb  pentane.pdb  propane.pdb
 > ~~~
 > {: .output}
 >
-> What is the output of:
+> What is the output of the following code?
 >
 > ~~~
-> for datafile in *.dat
+> for datafile in *.pdb
 > do
->     ls *.dat
+>     ls *.pdb
 > done
 > ~~~
 > {: .bash}
 >
-> Now, what is the output of:
+> Now, what is the output of the following code?
 >
 > ~~~
-> for datafile in *.dat
+> for datafile in *.pdb
 > do
 >	ls $datafile
 > done
 > ~~~
 > {: .bash}
 >
 > Why do these two loops give different outputs?
+>
+> > ## Solution
+> > The first code block gives the same output on each iteration through
+> > the loop.
+> > Bash expands the wildcard `*.pdb` within the loop body (as well as
```

Comment:
How about explicitly writing out the expanded loop?

Bash expands the wildcards before the loop starts, so the loop is equivalent to:

~~~
for datafile in cubane.pdb  ethane.pdb  methane.pdb  octane.pdb  pentane.pdb  propane.pdb
do
    ls cubane.pdb  ethane.pdb  methane.pdb  octane.pdb  pentane.pdb  propane.pdb
done
~~~